### PR TITLE
Added Node.js v10 package.

### PR DESCRIPTION
**Pull request by florianbehrens**
_Motivation:_  Added Node.js v10 package (Synology only provides <= v8).
_Linked issues:_  #3682 

### Checklist
- [x] Build rule `all-supported` completed successfully
- [x] Package upgrade completed successfully
- [x] New installation of package completed successfully

### CODE REVIEW

**Comment by florianbehrens** [comment]
I can confirm that the Node.js package runs on my DS216j. Is there anybody else who can test it on other Synology platforms?

My testcases where a small tool called `nasa-cli` and a short informal test of a running ioBroker application.

BTW: Is there any public CI system used by Synocommunity?

**Comment by florianbehrens** [comment]
I have no idea what "Package upgrade completed successfully" in the PR checklist means. Can somebody give some advice?

**Comment by MatthiasLohr** [comment]
Install an old version of the package. Try to upgrade to the new one. Does it succeed?

**Comment by florianbehrens** [comment]
Ok, thanks. Obviously there is no old version of the package. So marking the checkbox is fair?

**Comment by schumi2004** [comment]
I will test it later today since i need it for node-red
Node-red requires node.js higher then 6.x

Would it be an option to include node-red in the package or is that not possible?

**Comment by florianbehrens** [comment]
> Would it be an option to include node-red in the package or is that not possible?

Well, afaik node-red is available via `npm` which is part of the Node.js package. If it does not rely on binary dependencies as some NPM packages do (otherwise you would need a compiler on your target machine), the installation should be as easy as typing `npm i node-red`.

> Can't be compiled for Alpine (DS715)

Did you build it inside a Docker container based upon the updated Dockerfile?

**Comment by schumi2004** [comment]
Build inside Docker? I never compile packages inside Docker. I have a dedicated Ubuntu server were I compile packages in.

**Comment by florianbehrens** [comment]
@schumi2004 Well, the build process of Node.js is quite complex and I can't guarantee that it works on an arbitrary Ubuntu machine. The only thing I can say is that it builds for your architecture in the reference Docker image:

https://circleci.com/gh/florianbehrens/spksrc/108

**Comment by schumi2004** [comment]
@florianbehrens 
Okay great. 
Do you mind sharing the generated SPK since setting up Docker brakes Virtualbox for me. I need to do some research on how to setup a proper spksrc Docker environment first before i can compile ;)

Thanks in advance.

**Comment by florianbehrens** [comment]
@schumi2004 Yes I'm working on it. I was close to get it running on Travis but the build time limit hit me. Now I'm fighting with CircleCI, Docker, and GitHub Releases. Give me one more day.

**Comment by florianbehrens** [comment]
@schumi2004 Otherwise you could try to follow the hints in [this blogpost](https://chrislea.com/2018/08/20/cross-compiling-node-js-for-arm-on-ubuntu/) and give it another try on your Ubuntu machine.

**Comment by florianbehrens** [comment]
Here we are: Everyone is invited to test the packages available [here](https://github.com/florianbehrens/spksrc/releases/tag/build-node-v1) on his/her diskstation model.
They are built by [CircleCI](https://circleci.com/workflow-run/b7ed745f-29b9-49c8-a9ed-81fc7b2fc2f0) from branch `node-circleci-build` in my repo fork.

**Comment by schumi2004** [comment]
@florianbehrens You are awesome, it works perfect.
Also got Node-Red running which i need for Home Assistant which is my next step to install ;)

root@DS715:~# node-red
27 Jun 21:34:28 - [info] 

Welcome to Node-RED
===================

27 Jun 21:34:28 - [info] Node-RED version: v0.20.6
etc . .

**Comment by florianbehrens** [comment]
I added a few packages for `armv8` architectures, please see [here](https://github.com/florianbehrens/spksrc/releases/tag/build-node-v2).

**Comment by florianbehrens** [comment]
The "official" Node.js packages are called "Node.js vXY". My package currently is named just "node". Is it better to follow the Synology convention? That might result in a name clash as soon as Synology brings their own Node.js v10 package out. What do you think?

**Comment by florianbehrens** [comment]
@chickenandpork @ymartin59 Is there anything I can do to make progress on this?

File: cross/node/Makefile [hunk 1]
```diff
@@ -0,0 +1,53 @@
+PKG_NAME = node
+PKG_VERS = v10.15.3
+PKG_EXT = tar.gz
+PKG_DIST_NAME = $(PKG_NAME)-$(PKG_VERS).$(PKG_EXT)
+PKG_DIST_SITE = https://nodejs.org/dist/v10.15.3
```
Comment (ymartin59) [inline cross/node/Makefile]: Please replace version path part with $(PKG_VERS)

File: cross/node/PLIST.bak [hunk 2]
```diff
@@ -0,0 +1,10 @@
+bin:usr/local/bin/node
+bin:usr/local/bin/npm
```
Comment (ymartin59) [inline cross/node/PLIST.bak]: As far as there are too many lines in current PLIST, I prefer that version of PLIST file (with wildcard). Please use it instead, after removal of `usr/local` and replacement of `bin` by `rsc` for `npm` and `npx`.

**Comment by ymartin59** [comment]
Package name "node" is too short/generic. May you please rename it as "nodejs" for instance?

**Comment by florianbehrens** [comment]
@ymartin59 Thanks for your comments. Please see my changes and updated builds [here](https://github.com/florianbehrens/spksrc/releases/tag/build-node-v5).

**Comment by ymartin59** [comment]
I squashed commit to merge, published at beta: https://synocommunity.com/package/nodejs
Surprisingly, it gain the top rank for "longest build package", replacing mono .net runtime, but generated packages are far lighter (only 15 MiB instead of 180 MiB for mono)

**Comment by florianbehrens** [comment]
Great! Thanks!

**Pull request merged**